buildSrc/src/main/java/org/elasticsearch/gradle/NoticeTask.java

```diff
@@ -0,0 +1,128 @@
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.gradle;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+
+import org.gradle.api.DefaultTask;
+import org.gradle.api.tasks.InputFile;
+import org.gradle.api.tasks.OutputFile;
+import org.gradle.api.tasks.TaskAction;
+
+
+/**
+ * A task to create a notice file which includes dependencies' notices.
+ */
+public class NoticeTask extends DefaultTask {
+
+    private static final Charset UTF8 = Charset.forName("UTF-8");
+    @InputFile
+    private File inputFile = getProject().getRootProject().file("NOTICE.txt");
+
+    @OutputFile
+    private File outputFile = new File(getProject().getBuildDir(), "notices/" + getName() + "/NOTICE.txt");
+
+    /**
+     * Directories to include notices from
+     */
+    private List<File> licensesDirs = new ArrayList<>();
+
+    public NoticeTask() {
+        setDescription("Create a notice file from dependencies");
+        // Default licenses directory is ${projectDir}/licenses (if it exists)
+        File licensesDir = new File(getProject().getProjectDir(), "licenses");
+        if (licensesDir.exists()) {
+            licensesDirs.add(licensesDir);
+        }
+    }
+
+    /**
+     * Add notices from the specified directory.
+     */
+    public void setLicensesDir(File licensesDir) {
+        licensesDirs.add(licensesDir);
+    }
+
+    public void setInputFile(File inputFile) {
+        this.inputFile = inputFile;
+    }
+
+    @TaskAction
+    public void generateNotice() throws IOException {
+        StringBuilder output = new StringBuilder();
+        output.append(read(inputFile));
+        output.append("\n\n");
+        // This is a map rather than a set so that the sort order is the 3rd
+        // party component names, unaffected by the full path to the various files
+        Map<String, File> seen = new TreeMap<>();
+        for (File licensesDir : licensesDirs) {
+            File[] files = licensesDir.listFiles((file) -> file.getName().matches(".*-NOTICE\\.txt"));
+            if (files == null) {
+                continue;
+            }
+            for (File file : files) {
+                String name = file.getName().substring(0, file.getName().length() - "-NOTICE.txt".length());
+                if (seen.containsKey(name)) {
+                    File prevFile = seen.get(name);
+                    if (!read(prevFile).equals(read(file))) {
```

Comment:
We use `== False` throughout the codebase instead of `!` because the former is easier to spot visually.